lib/child-writer/index.js

```diff
@@ -170,14 +182,102 @@ exports.write = (image, drive, options) => {
     }
   }
 
+  // Timeout function that only runs if we exceed the specified timeout,
+  // terminating the server.
+  // @type {Timeout}
+  let terminationTimeout = null
+
+  // @type {Object} state
+  // @type {ChildProcess | undefined} [state.child]
+  // @type {Socket | undefined} [state.childSocket]
+  // @type {Boolean | undefined} [state.isChildRunning]
+  // @type {Boolean | undefined} [state.isChildSocketConnected]
+  const state = {}
+
+  /**
+   * @summary Finish the child writer
+   * @function
+   * @private
+   *
+   * @param {Number} code - exit code
+   * @returns {Void}
+   *
+   * @example
+   * finish(EXIT_CODES.SUCCESS)
+   */
+  const finish = (code) => {
+    terminateServer()
+    clearTimeout(terminationTimeout)
+
+    if (code === EXIT_CODES.CANCELLED) {
+      return emitter.emit('done', {
+        cancelled: true
+      })
+    }
+
+    // We shouldn't emit the `done` event manually here
+    // since the writer process will take care of it.
+    if (code === EXIT_CODES.SUCCESS || code === EXIT_CODES.VALIDATION_ERROR) {
+      return null
+    }
+
+    return emitError(new Error(`Child process exited with error code: ${code}`))
+  }
+
+  /**
+   * @summary Reset the timeout countdown
+   * @function
+   * @private
+   *
+   * @example
+   * resetTerminationTimeout()
+   */
+  const resetTerminationTimeout = () => {
+    clearTimeout(terminationTimeout)
+    terminationTimeout = setTimeout(() => {
+      const error = new Error('Write process timed out')
+      error.timeout = CHILD_WRITER_TIMEOUT_MS
+      error.isChildRunning = state.isChildRunning
+      error.childSocket = _.pick(state.childSocket, [
+        'connecting',
+        'destroyed',
+        'bytesRead',
+        'bytesWritten'
+      ])
+      error.childSocket.isConnected = state.isChildSocketConnected
+      error.child = _.pick(state.child, [ 'signalCode' ])
+      console.log(_.assign({}, error))
```

Comment:
Why copy the error's enumerable props when logging it? This will not copy over `.message`, `.stack` etc. And why are we `console.log()`ing this at all?, it'll be emitted through the `timeout` event, right? Or is this just a debug leftover? So many questions! :D

Reply:
Hah yeah, debugging leftovers – let me remove that.